Lib/ipaddress.py

```diff
@@ -1317,36 +1338,21 @@ def is_link_local(self):
 class IPv4Interface(IPv4Address):
 
     def __init__(self, address):
-        if isinstance(address, (bytes, int)):
-            IPv4Address.__init__(self, address)
-            self.network = IPv4Network(self._ip)
-            self._prefixlen = self._max_prefixlen
-            return
+        addr_prefix = self._get_addr_prefix_tuple(address)
+        addr, (self.netmask, self._prefixlen) = addr_prefix
+        IPv4Address.__init__(self, addr._ip)
 
-        if isinstance(address, tuple):
-            IPv4Address.__init__(self, address[0])
-            if len(address) > 1:
-                self._prefixlen = int(address[1])
-            else:
-                self._prefixlen = self._max_prefixlen
-
-            self.network = IPv4Network(address, strict=False)
-            self.netmask = self.network.netmask
-            self.hostmask = self.network.hostmask
-            return
+    @functools.cached_property
+    def network(self):
+        return IPv4Network((self._ip, self._prefixlen), strict=False)
 
-        addr = _split_optional_netmask(address)
-        IPv4Address.__init__(self, addr[0])
-
-        self.network = IPv4Network(address, strict=False)
-        self._prefixlen = self.network._prefixlen
-
-        self.netmask = self.network.netmask
-        self.hostmask = self.network.hostmask
+    @functools.cached_property
+    def hostmask(self):
```

Comment:
This seems like a new property as per current docs at https://docs.python.org/3.8/library/ipaddress.html#ipaddress.IPv4Interface . Should this be documented?